tests/acceptance-tests/src/acceptance/http_forwarded_for_test.go

```diff
@@ -0,0 +1,62 @@
+package acceptance_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+
+	"github.com/cloudfoundry-incubator/cf-test-helpers/cf"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/generator"
+	"github.com/cloudfoundry-incubator/cf-test-helpers/helpers"
+
+	"encoding/json"
+	"net"
+	"strings"
+)
+
+type ResponseJson struct {
+	Header struct {
+		X_Forwarded_For   []string `json:"X-Forwarded-For"`
+		X_Forwarded_Proto []string `json:"X-Forwarded-Proto"`
+	} `json:"Header"`
+}
+
+var headers ResponseJson
+
+var _ = BeforeEach(func() {
+	appName := generator.PrefixedRandomName("CATS-APP-")
+	Expect(cf.Cf(
+		"push", appName,
+		"-b", config.GoBuildpackName,
+		"-p", "../../example_apps/print_request_headers",
+		"-d", config.AppsDomain,
+		"-c", "./bin/debug_app; sleep 1; echo 'done'",
+	).Wait(CF_PUSH_TIMEOUT)).To(Exit(0))
+
+	jsonData := helpers.CurlAppRoot(appName)
+	headers = ResponseJson{}
+	json.Unmarshal([]byte(jsonData), &headers)
+})
+
+var _ = Describe("HTTP forwarded headers are set correctly", func() {
+
+	// Validate that the upstream IP is _not_ on our network
+	It("for X-Forwarded-For", func() {
+		remoteIP := strings.Split(headers.Header.X_Forwarded_For[0], ",")[0]
+		parsedRemoteIP := net.ParseIP(remoteIP)
+		_, infra_cidr, _ := net.ParseCIDR("10.0.0.0/8")
+
+		Expect(infra_cidr.Contains(parsedRemoteIP)).To(Equal(false))
```

Comment:
I think the following has the same effect but it slightly easier to read:
```go
Expect(infra_cidr).NotTo(ContainElement(parsedRemoteIP))
```

ref: https://onsi.github.io/gomega/#containelementelement-interface